compactor2/src/driver.rs

```diff
@@ -79,6 +79,95 @@ async fn compact_partition(
     scratchpad.clean().await;
 }
 
+// Main function to compact files of a partitions.
```

Comment:
I think it would help here 
1. Make these doc comments (`///`)
2. Add a description of the output state after this function returns

For example (I am not sure if this is correct, but it is the kind of description I think that would help)

```suggestion
/// Main function to compact files of a single partition.
/// 
/// ///
/// When this function returns, the parquet files that existed at the start of the call will 
/// be completely non-overlapping L1 [`CompactionLevel:: FileNonOverlapped`] or
/// L2 [`CompactionLevel::Final`] files. Any L0 [`CompactionLevel::Initial`] files will be 
/// marked for deletion in the catalog (soft deletion)
///
/// Any new parquet files created while this function is running are not considered
///  (and thus may overlap).
```